nix/UPGRADE.md

```diff
@@ -0,0 +1,94 @@
+# Checklist for upgrading Nix dependencies
+
+The Nix dependencies of PostgREST should be updated regularly, in most cases it
+should be a very simple operation.
+
+```bash
+# Update pinned version of Nixpkgs
+nix-shell --run nixpkgs-upgrade > nix/nixpkgs-version.nix
+
+# Verify that everything builds
+nix-build
+```
+
+The following checklist guides you through the complete process in more detail.
+
+## Upgrade the pinned version of `nixpkgs`
+
+The pinned version of [`nixpkgs`](https://github.com/NixOS/nixpkgs) is defined
+in [`nix/nixpkgs-version.nix`](nixpkgs-version.nix). The pin refers directly to
+a GitHub tarball for the given revision, which is more efficient than pulling
+the complete Git repository. To upgrade it to the current `master` of
+`nixpkgs`, you can use a small utility script defined in
+[`nix/nixpkgs-update.nix`](nixpkgs-update.nix):
+
+```bash
+# From the root of the repository, enter nix-shell
+nix-shell
+
+# Run the utility script to pin the latest revision in master
+nixpkgs-upgrade > nix/nixpkgs-version.nix
+
+# Exit the nix-shell with Ctrl-d
+
+# Alternatively, in one step
+nix-shell --run nixpkgs-upgrade > nix/nixpkgs-version.nix
```

Comment:
This one gives me the following error in and outside of  `nix-shell`:

```
error: while evaluating anonymous function at /home/steve-chavez/Projects/Haskell/postgrest/shell.nix:12:1, called from undefined position:
while evaluating the attribute 'pkgs' at /home/steve-chavez/Projects/Haskell/postgrest/default.nix:80:10:
syntax error, unexpected $end, at /home/steve-chavez/Projects/Haskell/postgrest/nix/nixpkgs-version.nix:1:1
```

The `nix/nixpkgs-version.nix` file contents get empty. I guess that's what the `unexpected $end` error means.

The `nixpkgs-upgrade > nix/nixpkgs-version.nix` version inside `nix-shell` works good though. 

Is something missing on the command?

If the fix is not easy, maybe we can omit the one liner.

Reply:
Hm, weird - just tested it and it works for me. Maybe it's the nix-shell version? I'm on 2.3.7

But I guess it doesn't hurt to take out that line for now :-)

Reply:
Ah, yes. I'm on `2.2.2`. It's like bash redirection doesn't work on that version. Maybe a bug that got fixed.